src/libraries/ChainlinkPairDetails.sol

```diff
@@ -0,0 +1,85 @@
+import {ChainlinkOracleImpl} from "../chainlink/oracle/ChainlinkOracleImpl.sol";
+import {ChainlinkPath} from "./ChainlinkPath.sol";
+
+/// @title Chainlink PairDetails Library
+/// @author 0xSplits
+/// @notice Setters & getters for quote pairs' chainlink details
+library ChainlinkPairDetails {
+    /// -----------------------------------------------------------------------
+    /// libraries
+    /// -----------------------------------------------------------------------
+
+    using ChainlinkPath for bytes;
+
+    /// -----------------------------------------------------------------------
+    /// errors
+    /// -----------------------------------------------------------------------
+    error InvalidFeed_StaleAfter();
+    error InvalidFeed_Decimals();
+    error InvalidFeed_Mul();
+    error InvalidFeed_Empty();
+
+    /// set pair details
+    function _set(
+        mapping(address => mapping(address => ChainlinkOracleImpl.PairDetail)) storage self,
+        function (address) internal view returns (address) convert,
+        ChainlinkOracleImpl.SetPairDetailParams[] calldata params_
+    ) internal {
+        uint256 length = params_.length;
+        for (uint256 i; i < length;) {
+            _validate(params_[i]);
+            _set(self, convert, params_[i]);
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// set pair details
+    function _set(
+        mapping(address => mapping(address => ChainlinkOracleImpl.PairDetail)) storage self,
+        function (address) internal view returns (address) convert,
+        ChainlinkOracleImpl.SetPairDetailParams calldata params_
+    ) internal {
+        ConvertedQuotePair memory cqp = params_.quotePair._convert(convert);
+        self[cqp.cBase][cqp.cQuote] = params_.pairDetail;
+        ChainlinkOracleImpl.PairDetail memory pairDetail = params_.pairDetail;
+        pairDetail.inverted = !pairDetail.inverted;
+        self[cqp.cQuote][cqp.cBase] = pairDetail;
+    }
+
+    /// get pair details
+    function _get(
+        mapping(address => mapping(address => ChainlinkOracleImpl.PairDetail)) storage self,
+        function (address) internal view returns (address) convert,
+        QuotePair calldata quotePair_
+    ) internal view returns (ChainlinkOracleImpl.PairDetail memory) {
+        return _get(self, quotePair_._convert(convert));
+    }
+
+    function _get(
+        mapping(address => mapping(address => ChainlinkOracleImpl.PairDetail)) storage self,
+        ConvertedQuotePair memory quotePair_
+    ) internal view returns (ChainlinkOracleImpl.PairDetail memory) {
+        return self[quotePair_.cBase][quotePair_.cQuote];
+    }
+
+    function _validate(ChainlinkOracleImpl.SetPairDetailParams calldata params_) private view {
+        ChainlinkOracleImpl.Feed[] memory feed = params_.pairDetail.path.getFeeds();
+
+        uint256 length = feed.length;
+        if (length == 0) revert InvalidFeed_Empty();
```

Comment:
if someone sets up a pair with a feed but later wants to remove it from the oracle, i believe this will prevent them from being able to do so